ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x067EA660>, trait: thresholds, old value: None, new value: 0.0 inf 0.0 inf 0.0 inf 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 350, in apply_thresholds
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x067EA660>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 372, in update_binning
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x067EA660>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 368, in discard_data_on_E
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x067EA660>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 359, in discard_data_on_S
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x067EA660>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 376, in update_weighting
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.controller.PRHController object at 0x0670E2D0>, trait: fitting, old value: None, new value: <alex.functions.GuessesToFit object at 0x06815DE0>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 210, in redraw_plot
    self.evss_view.update()
AttributeError: 'NoneType' object has no attribute 'update'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.views.EvsSView object at 0x0A7606F0>, trait: rectangle, old value: None, new value: <undefined>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 221, in add_guess_rect
    x1 = rect[0][0]
TypeError: '_Undefined' object does not support indexing
WARNING:enaml.session:Invalid object id sent to Session: o_gy:clicked
ERROR:__main__:Uncaught exception: name 'PopUp' is not defined
None
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0B0641B0>, trait: thresholds, old value: None, new value: 0.0 inf 0.0 inf 0.0 inf 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 350, in apply_thresholds
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0B0641B0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 372, in update_binning
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0B0641B0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 368, in discard_data_on_E
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0B0641B0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 359, in discard_data_on_S
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0B0641B0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 376, in update_weighting
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.controller.PRHController object at 0x0B0384B0>, trait: fitting, old value: None, new value: <alex.functions.GuessesToFit object at 0x0B06A2A0>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 210, in redraw_plot
    self.evss_view.update()
AttributeError: 'NoneType' object has no attribute 'update'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.views.EvsSView object at 0x0B06ACF0>, trait: rectangle, old value: None, new value: <undefined>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 221, in add_guess_rect
    x1 = rect[0][0]
TypeError: '_Undefined' object does not support indexing
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0B0641B0>, trait: updated, old value: <undefined>, new value: True
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\views.py", line 190, in update
    self.plot()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\views.py", line 216, in plot
    self.format()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\views.py", line 317, in format
    magnitude_E = int(np.log10(self.ax_E.yaxis.get_data_interval()[1]))
ValueError: cannot convert float NaN to integer
WARNING:enaml.session:Invalid object id sent to Session: o_w2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_r2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_j2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_d2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_d2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_j2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_d2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_j2:changed
WARNING:enaml.session:Invalid object id sent to Session: o_j2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_d2:changed
WARNING:enaml.session:Invalid object id sent to Session: o_d2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_d2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_j2:clicked
WARNING:enaml.session:Invalid object id sent to Session: o_d2:clicked
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0ABEB8A0>, trait: thresholds, old value: None, new value: 0.0 inf 0.0 inf 0.0 inf 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 350, in apply_thresholds
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0ABEB8A0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 372, in update_binning
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0ABEB8A0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 368, in discard_data_on_E
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0ABEB8A0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 359, in discard_data_on_S
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0ABEB8A0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 376, in update_weighting
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.controller.PRHController object at 0x067EA0F0>, trait: fitting, old value: None, new value: <alex.functions.GuessesToFit object at 0x0BC120F0>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 210, in redraw_plot
    self.evss_view.update()
AttributeError: 'NoneType' object has no attribute 'update'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.views.EvsSView object at 0x0A978B40>, trait: rectangle, old value: None, new value: <undefined>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 221, in add_guess_rect
    x1 = rect[0][0]
TypeError: '_Undefined' object does not support indexing
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0A2F2BD0>, trait: thresholds, old value: None, new value: 0.0 inf 0.0 inf 0.0 inf 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 350, in apply_thresholds
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0A2F2BD0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 372, in update_binning
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0A2F2BD0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 368, in discard_data_on_E
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0A2F2BD0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 359, in discard_data_on_S
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x0A2F2BD0>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 376, in update_weighting
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.controller.PRHController object at 0x0ABF38D0>, trait: fitting, old value: None, new value: <alex.functions.GuessesToFit object at 0x0BB25570>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 210, in redraw_plot
    self.evss_view.update()
AttributeError: 'NoneType' object has no attribute 'update'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.views.EvsSView object at 0x0B38EC00>, trait: rectangle, old value: None, new value: <undefined>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 221, in add_guess_rect
    x1 = rect[0][0]
TypeError: '_Undefined' object does not support indexing
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x10D85C60>, trait: thresholds, old value: None, new value: 0.0 inf 0.0 inf 0.0 inf 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 350, in apply_thresholds
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x10D85C60>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 372, in update_binning
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x10D85C60>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 368, in discard_data_on_E
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x10D85C60>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 359, in discard_data_on_S
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x10D85C60>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 376, in update_weighting
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.controller.PRHController object at 0x0AA6A060>, trait: fitting, old value: None, new value: <alex.functions.GuessesToFit object at 0x0B49B450>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 210, in redraw_plot
    self.evss_view.update()
AttributeError: 'NoneType' object has no attribute 'update'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.views.EvsSView object at 0x0AB21960>, trait: rectangle, old value: None, new value: <undefined>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 221, in add_guess_rect
    x1 = rect[0][0]
TypeError: '_Undefined' object does not support indexing
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x11318F30>, trait: thresholds, old value: None, new value: 0.0 inf 0.0 inf 0.0 inf 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 350, in apply_thresholds
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x11318F30>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 372, in update_binning
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x11318F30>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 368, in discard_data_on_E
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x11318F30>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 359, in discard_data_on_S
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.models.EvsS object at 0x11318F30>, trait: options, old value: None, new value: False 101 101 0.0 1.0 0.0 1.0
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 376, in update_weighting
    self.update()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 448, in update
    self._accurate_E()
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\models.py", line 476, in _accurate_E
    DA_bkg_corr = self.data['F_Dexc_Aem'] - self.corr.bkg_DA*self.data['Tau']*1000
AttributeError: 'NoneType' object has no attribute 'bkg_DA'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.controller.PRHController object at 0x0B1A2420>, trait: fitting, old value: None, new value: <alex.functions.GuessesToFit object at 0x0AB20960>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 210, in redraw_plot
    self.evss_view.update()
AttributeError: 'NoneType' object has no attribute 'update'
ERROR:traits:Exception occurred in traits notification handler for object: <alex.views.EvsSView object at 0x0AB4C900>, trait: rectangle, old value: None, new value: <undefined>
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 520, in _dispatch_change_event
    self.dispatch( handler, *args )
  File "C:\Python27\lib\site-packages\traits\trait_notifiers.py", line 483, in dispatch
    handler( *args )
  File "X:\Data\Desktop\ALEX-Suite 1.0\alex\controller.py", line 221, in add_guess_rect
    x1 = rect[0][0]
TypeError: '_Undefined' object does not support indexing
